harmonic/troubleshooting.md

```diff
@@ -0,0 +1,178 @@
+
+There can be out of memory issue during the compilation of Gazebo, especially during
+compilation of gz-physics. To prevent out of memory issues you can restrict the number of jobs:
+
+```bash
+MAKEFLAGS="-j<Number of jobs> " colcon build --executor sequential
+```
+
+### Problems with dual Intel and Nvidia GPU systems
+
+If you are using a dual Intel/Nvidia system it could be the case that the
+simulator is being run under Intel instead of using the Nvidia GPU. Bugs can
+vary but there could problems with shadows, incorrect laser scans or other
+rendering related issues.
+
+#### prime-select command line tool
+
+Hybrid Intel/Nvidia systems can be configured using the command line tool prime-select.
+One option is to use always Nvidia:
+
+    sudo prime-select nvidia
+    # logout user session and login again
+
+Other option is to configure the render offload for OpenGL applications to use
+Nvidia. This means that your X screen and all normal applications are handled
+by the Intel GPU, but all OpenGL applications that you start from the terminal
+(including Gazebo) are rendered on the Nvidia GPU.
+
+    # place the lines in your .bashrc if you want the change to be permanent
+    export __NV_PRIME_RENDER_OFFLOAD=1
+    export __GLX_VENDOR_LIBRARY_NAME=nvidia
+    # logout user session and login again
+
+#### nvidia-settings GUI tool
+
+nvidia-settings is a GUI program that helps to configure the options for the Nvidia
+graphic cards and includes some controls for hybrid Intel/Nvidia:
+
+The section "PRIME Profiles" can be used to select that the Nvidia card controls
+all the GUI applications by selecting "NVIDIA (Performance Mode)".
+
+The "Application Profiles" can control the use of the Nvidia GPU per application.
+
+### Unable to create the rendering window
+If you're getting errors like "Unable to create the rendering window", it could
+mean you're using an old OpenGL version. Gazebo Sim uses the Ogre 2
+rendering engine by default, which requires an OpenGL version higher than 3.3.
+
+This can be confirmed by checking the Ogre 2 logs at `~/.gz/rendering/ogre2.log`,
+which should have an error like:
+
+    "OGRE EXCEPTION(3:RenderingAPIException): OpenGL 3.3 is not supported. Please update your graphics card drivers."
+
+You can also check your OpenGL version running:
+
+    glxinfo | grep "OpenGL version"
+
+You should be able to use Ogre 1 without any issues however. You can check if
+that's working by running a world which uses Ogre 1 instead of Ogre 2, such as:
+
+    gz sim -v 3 lights.sdf
+
+If that loads, you can continue to use Gazebo Sim with Ogre 1, just use the
+`--render-engine ogre` option.
+
+To enable Ogre 2 support, you'll need to update your computer's OpenGL version.
+As suggested on the Ogre logs, this may require updating your graphics card
+drivers.
+
+The Ogre 2 debs from the osrfoundation repository are built from a fork of
+Ogre's `v2-3` branch with changes needed for deb packaging and allowing it to
```

Comment:
The garden version has `v2-1`. I think we need to update that as well.

Reply:
done. [937916a](https://github.com/gazebosim/docs/pull/324/commits/937916a493ee0ed5f721e5ab648f62c5d4717a3d)